pkg/supervisor/supervisor_nix.go

```diff
@@ -0,0 +1,98 @@
+//go:build !windows
+// +build !windows
+
+/*
+Copyright 2022 k0s authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package supervisor
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+)
+
+const (
+	exitCheckInterval = 200 * time.Millisecond
+)
+
+// killPid signals SIGTERM to a PID and if it's still running after
+// s.TimeoutStop sends SIGKILL.
+func (s *Supervisor) killPid(pid int) error {
+	if s.KillFunction == nil {
+		s.KillFunction = syscall.Kill
+	}
+	// Kill the process pid
+	if err := s.KillFunction(pid, syscall.SIGTERM); err != nil {
+		return fmt.Errorf("failed to send SIGTERM to pid %d: %s", pid, err)
+	}
+
+	deadline := time.Now().Add(s.TimeoutStop)
+	for time.Now().Before(deadline) {
+		if !s.isRunning(pid) {
+			return nil
+		}
+		time.Sleep(exitCheckInterval)
+	}
+
+	if err := s.KillFunction(pid, syscall.SIGKILL); err != nil {
+		return fmt.Errorf("failed to send SIGKILL to pid %d: %s", s.cmd.Process.Pid, err)
+	}
+	return nil
+}
+
+func (s *Supervisor) isRunning(pid int) bool {
+	_, err := os.Stat(filepath.Join(s.ProcFSPath, strconv.Itoa(pid)))
```

Comment:
`s.KillFunction(pid, 0)` can also be used to test if process is running.

Reply:
I like the idea of having the same code handling both the test and the real cases, however that's certainly cleaner and allows us to remove Supervisor.ProcFSPath...